content/blog/esc-secret-rotation-with-iac/index.md

```diff
@@ -0,0 +1,162 @@
+---
+title: "Secret Rotation with Pulumi ESC"
+date: 2024-12-02T11:23:17-08:00
+draft: false
+meta_desc: "Extending ESC to perform automated secret rotation."
+meta_image: meta.png
+authors:
+    - claire-gaestel
+tags:
+  - esc
+  - secrets
+---
+
+Managing secrets in modern cloud applications is challenging, particularly when it comes to rotation policies. While dynamic secrets (like AWS IAM temporary credentials) handle this automatically, many systems still rely on static secrets that need periodic rotation.
+
+Static secrets, like database passwords or API keys, [should be rotated regularly to maintain security](https://cheatsheetseries.owasp.org/cheatsheets/Secrets_Management_Cheat_Sheet.html#272-rotation), but services depending on these secrets need time to transition to new credentials to avoid downtime. This makes rotating credentials error-prone, and often forgotten.
```

Comment:
```suggestion
Static secrets, like database passwords or API keys, [should be rotated regularly to maintain security](https://cheatsheetseries.owasp.org/cheatsheets/Secrets_Management_Cheat_Sheet.html#272-rotation), and services depending on these secrets need time to transition to new credentials to avoid downtime. This makes rotating credentials error-prone, and often forgotten.
```